packages/@aws-cdk/core/lib/construct.ts

```diff
@@ -438,6 +438,15 @@ export class ConstructNode {
     return ret;
   }
 
+  /**
+   * Remove the child with the given name, if present.
+   *
+   * It is not an error if a child with the given name does not exist.
+   */
+  public removeChild(childName: string) {
```

Comment:
Maybe rename to `tryRemoveChild` to indicate it won't throw, and return a boolean indicating if the child was actually removed or not.

I am a bit worried by potential abuse of this API. Maybe we can make it a bit safer and only allow this during "prepare" somehow? What do you think? Can you please also mark this as `@experimental`?

This also resolves https://github.com/aws/aws-cdk/issues/1408

Reply:
I'm not too worried about misuse. The methods on `node` are pretty much "plumbing sticking out" anyway, not sure they're part of people's day-to-day operations with the API.

I would hope that people venturing there will be aware they're messing with internals which can do what they want, or have unintentional effects they themselves will be responsible for cleaning up. Not sure that limiting to `prepare()` is necessary.